packages/terra-action-header/src/ActionHeader.jsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconLeft from 'terra-icon/lib/icon/IconLeft';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import ResponsiveElement from 'terra-responsive-element';
+import 'terra-base/lib/baseStyles';
+import ActionHeaderContainer from './_ActionHeaderContainer';
+import styles from './ActionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Text to be displayed as the title in the header bar.
+   */
+  title: PropTypes.string,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=1`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  /**
+   * Callback function for when the close button is clicked. The back button will not display if this is not set.
+   * On small viewports a back button will be displayed instead of a close button when a separate onBack callback is not set.
+   */
+  onClose: PropTypes.func,
+  /**
+   * Callback function for when the back button is clicked. The back button will not display if this is not set.
+   */
+  onBack: PropTypes.func,
+  /**
+   * A Boolean indicating if close button should be displayed on small viewports when separate onBack callback is not set.
+   */
+  keepCloseButton: PropTypes.bool,
+  /**
+   * Child element to be displayed on the right end of the header.
+   */
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  title: '',
```

Comment:
I think it's better to make the title undefined as opposed to an empty string to default, unless we noted in a tech design or decision document.

Reply:
Updated [here](https://github.com/cerner/terra-core/commit/5b230102a59e8b12ddab3baf4971a109535b0884).